security/pkg/pki/ca/ca.go

```diff
@@ -232,15 +260,49 @@ func NewSelfSignedDebugIstioCAOptions(rootCertFile string, caCertTTL, defaultCer
 	return caOpts, nil
 }
 
+func specifySelfSignedAlgorithmIstioCAOptions(opts *IstioCAOptions, caRSAKeySize int, ecSigAlg, eccCurve string) error {
+	switch opts.AlgorithmType {
+	case util.RsaAlg:
+		opts.CARSAKeySize = caRSAKeySize
+	case util.EcAlg:
+		opts.ECSigAlg = util.SupportedECSignatureAlgorithms(ecSigAlg)
+		opts.ECCCurve = util.SupportedEllipticCurves(eccCurve)
```

Comment:
nit: what happens if the eccCurve is set to something unsupported? This would eventually fail when we try to create the DNS cert, but think it would make sense to fail early in ```createIstioCA```

Reply:
Added a check to make sure both are supported.